bot/cogs/reddit.py

```diff
@@ -42,16 +43,22 @@ def __init__(self, bot: Bot):
         if params is None:
             params = {}
 
-        response = await self.bot.http_session.get(
-            url=f"{self.URL}/{route}.json",
-            headers=self.HEADERS,
-            params=params
-        )
-
-        content = await response.json()
-        posts = content["data"]["children"]
+        url = f"{self.URL}/{route}.json"
+        for _ in range(self.MAX_FETCH_RETRIES):
+            response = await self.bot.http_session.get(
+                url=url,
+                headers=self.HEADERS,
+                params=params
+            )
+            if response.status == 200 and response.content_type == 'application/json':
+                # Got appropriate response - process and return.
+                content = await response.json()
+                posts = content["data"]["children"]
+                return posts[:amount]
+            await asyncio.sleep(3)
```

Comment:
Putting a space here will make it slightly more readable

```suggestion

            await asyncio.sleep(3)
```